### PR TITLE
Closes #735: Moving dataset from UI

### LINKED ISSUE #735: Implement moving dataset to other dataverse on superadmin dashboard

### PR DESCRIPTION
**Pull request by wtr-icm**

### CODE REVIEW

File: dataverse-persistence/src/main/resources/Bundle_en.properties [hunk 1]
```diff
@@ -2607,3 +2607,37 @@ facets.search.fieldtype.dateOfDeposit.label=Deposit Date
 
 #privacy-policy.xhtml
 privacy.policy.page.title=Privacy policy
+
```
Comment (wtr-icm) [inline dataverse-persistence/src/main/resources/Bundle_en.properties]: Both bundles are not in the final stage at the moment.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardDatamovePage.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
+package edu.harvard.iq.dataverse.dashboard;
```
Comment (wtr-icm) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardDatamovePage.java:1]: This is moved from Dataverse 4.11 with a heavy refactor

File: dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                                                </f:facet>
+                                                <h:outputText value="#{srcDS.owner.alias}"/>
+                                            </p:column>
+                                            <p:ajax process="@this" event="itemSelect" />
+                                            <p:ajax process="@this" event="itemUnselect" />
+                                        </p:autoComplete>
+                                        <script>
+                                            handle_dropdown_popup_scroll();
+                                        </script>
+                                        <p:message for="sourceDatasetMenu"/>
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml]: In most places we show validation messages as text only, so I would add `display="text"` argument for this `p:message`

File: dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                                                </f:facet>
+                                                <h:outputText value="#{destDV.alias}"/>
+                                            </p:column>
+                                            <p:ajax process="@this" event="itemSelect" />
+                                            <p:ajax process="@this" event="itemUnselect" />
+                                        </p:autoComplete>
+                                        <script>
+                                            handle_dropdown_popup_scroll();
+                                        </script>
+                                        <p:message for="destinationDataverseMenu"/>
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml]: same comment as above

File: dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+                    <div class="col-sm-12 button-block">
+                        <p:commandButton id="move" styleClass="btn btn-default" 
+                                       value="#{bundle['dashboard.datamove.button.move']}"
+                                       update="datamove-form"
+                                       validateClient="true"
+                                       oncomplete="if (args &amp;&amp; !args.validationFailed) PF('datamoveConfirmation').show();">
+                            <!-- if (args &amp;&amp; !args.validationFailed) 
+                             prevents the confirm dialog to pop up (client side) if validation fails -->
+                            <f:param name="DO_DATAMOVE_VALIDATION" value="true"/>
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml]: Not sure if this param is ever used

File: dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                        </p:commandButton>
+                        <p:button styleClass="btn btn-link" outcome="dashboard" value="#{bundle['cancel']}">
+                            <!-- dataverse.id resolves to null !
+                                 <f:param name="dataverseId" value="#{dataverse.id}" />
+                            -->
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/dashboard-datamove.xhtml]: Not helpful comment - I would get rid of it

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardDatamovePage.java [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+            JsfHelper.addFlashErrorMessage(getStringFromBundle("dashboard.datamove.empty.fields"));
+            return;
+        }
+
+        Summary summary = new Summary().addParameter(sourceDataset.getDisplayName())
+                .addParameter(extractSourcePersistentId())
+                .addParameter(targetDataverse.getName());
+
+        try {
+            DataverseRequest dataverseRequest = new DataverseRequest(authenticatedUser, request);
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardDatamovePage.java]: You can use `DataverseRequestServiceBean` to obtain this object. That way you can omit injecting `HttpServletRequest` and saving `authenticatedUser` as a class field

**Review by madryk — CHANGES_REQUESTED** [review]

**Review by madryk — APPROVED** [review]

**Pull request merged**